app/src/main/java/com/example/dndwiki/repository/SpellRepository.kt

```diff
@@ -33,11 +35,20 @@ class SpellRepository(
     }
 
     private suspend fun saveAllSpellDetails() {
-        api.getSpells().forEach {
-            db.saveSpellDetails(api.getSpellDetails(it.index))
+        val spells = api.getSpells()
+        val spellDetails = mutableListOf<SpellDetails>()
+        withContext(Dispatchers.IO) {
+            spells.forEach {
+                launch {
+                    val api = async { api.getSpellDetails(it.index) }
```

Comment:
No need for the async, you're already in launch

you can just call the api from here and get the result normally